web/app/js/index.js

```diff
@@ -32,29 +31,62 @@ if (proxyPathMatch) {
   pathPrefix = proxyPathMatch[0];
 }
 
-const context = {
-  ...appData,
-  api: ApiHelpers(pathPrefix),
-  pathPrefix: pathPrefix,
-  productName: "Linkerd"
-};
+let selectedNamespace = "default";
+let pathArray = window.location.pathname.split("/");
+if (pathArray[0] === "" && pathArray[1] === "namespaces" && pathArray[2]) {
+  selectedNamespace = pathArray[2];
+}
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      ...appData
+    };
 
-const theme = createMuiTheme(dashboardTheme);
+    this.state.api = ApiHelpers(pathPrefix);
+    this.state.pathPrefix = pathPrefix;
+    this.state.productName = "Linkerd";
+    this.state.selectedNamespace = selectedNamespace;
```

Comment:
curious: in other places in the code i see us doing `this.state = this.getInitialState(...` or `this.state = {...`. is this a relevant convention?

Reply:
hey @siggy, see my note in the original PR message about the lint error vs the React Context docs. i think there must be a better solution than defining these variables line by line, but I'm prioritizing the UI in this PR.